src/volatile_memory.rs

```diff
@@ -312,15 +315,37 @@ pub struct PtrGuard {
     _slice: MmapXenSlice,
 }
 
+#[cfg(all(feature = "xen", unix))]
+impl PtrGuard {
+    const PROT_READ: i32 = libc::PROT_READ;
+    const PROT_WRITE: i32 = libc::PROT_WRITE;
+
+    fn map(
+        mmap: Option<&MmapInfo>,
+        addr: *mut u8,
+        prot: i32,
```

Comment:
Maybe it's simpler to change the `prot` argument to just be `read: bool`, then inside the body of this function we do `let prot = if read { libc::PROT_READ } else { libc::PROT_WRITE }`? (or even just have that in the old `cfg`'d block).

Reply:
Sure.